src/middleware/attachRedisClient.ts

```diff
@@ -1,12 +1,17 @@
-import { createClient } from 'redis';
+import { createClient, RedisClientType } from 'redis';
+import { NextFunction, Request, Response } from 'express';
 import { REDIS_HOST, REDIS_PORT, REDIS_URL, REDIS_USERNAME, REDIS_PASSWORD } from '../config';
 
-const afterResponse = (redisClient) => {
+interface RedisClientRequest extends Request {
+  redisClient: RedisClientType;
+}
+
+const afterResponse = (redisClient: RedisClientType) => {
   try {
     if (redisClient) {
       redisClient.quit();
     }
-  } catch (err) {
+  } catch (err: any) {
```

Comment:
we don't want to use `any`, rather it should be a defined or expected type like `Error`